tools/src/main/java/org/aya/util/ArrayUtil.java

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2020-2021 Yinsen (Tesla) Zhang.
+// Use of this source code is governed by the MIT license that can be found in the LICENSE.md file.
+package org.aya.util;
+
+import kala.tuple.Tuple;
+import kala.tuple.Tuple2;
+import kala.tuple.Tuple3;
+import org.jetbrains.annotations.NotNull;
+
+import java.lang.reflect.Array;
+import java.util.function.Function;
+
+@SuppressWarnings("unchecked")
+public interface ArrayUtil {
```

Comment:
I think we can use `ArraySeq.warp().view()` to implement these methods

Reply:
For traversal ones (like `identical`), yes, but for mapping ones, `ArraySeq.wrap().view()` creates a `SeqView`, but these methods create an array.

I propose deleting them if we decide to close #189 as wontfix.

Reply:
We can discuss #189 later when JEP 405 becomes `Delivered`. And I really don't like manipulating with raw arrays (they give a sense of unsafe) 😭

Reply:
Me either. That's why I changed my mind. Arrays are _very bad_ programming interfaces

Reply:
I'm not using these. I just thought that these _might_ be useful somewhere, so I put then into tools in case we need them in the future... And these are general enough to have a potential usage